WordPress/src/main/java/org/wordpress/android/ui/posts/PostsListActivity.kt

```diff
@@ -43,13 +47,18 @@ class PostsListActivity : AppCompatActivity(),
     private lateinit var site: SiteModel
     private lateinit var viewModel: PostListMainViewModel
 
+    private lateinit var authorSelectionAdapter: AuthorSelectionAdapter
+    private lateinit var authorSelection: AppCompatSpinner
+
     private lateinit var postsPagerAdapter: PostsPagerAdapter
     private lateinit var pager: ViewPager
     private lateinit var fab: FloatingActionButton
 
     private val currentFragment: PostListFragment?
         get() = postsPagerAdapter.getItemAtPosition(pager.currentItem)
 
+    private var updatingUI = false
```

Comment:
Is this variable really necessary? Since both `OnPageChangeListener` callbacks and `viewState` observer has to be called on the main thread, there should be no way to for the page to be changed while the author adapter is being updated. Am I maybe missing something?

Just as a note, if we did need a variable that is used across threads, we'd need a thread safe value which `boolean` is not.

Reply:
This is used to stop the `onPageSelected` event from firing and posting the same ui state again when we update our pager adapter. Which is triggered when we set the `currentItem`.

Given it's limited use I didn't believe it was necessary to use something more complicated to ensure thread safety.

If it is an issue, I would prefer to remove it and have the same state posted twice than make this more complicated with thread safe code.

Reply:
Oh I see what the problem is, it's not about threading, so please disregard that part of my comment. I _think_ what we'll want to do is to remove the listener first, create and set the adapter and then re-add the listener, however I'd like to get @malinajirka's opinion on it before we proceed.

Reply:
Ohh, I understand the usage now. 

>I think what we'll want to do is to remove the listener first, create and set the adapter and then re-add the listener,

This sounds like a good solution to me. 

>If it is an issue, I would prefer to remove it and have the same state posted twice than make this more complicated with thread safe code.

💯

SideNote: I think we should use boolean flags as a last resort. Nevertheless if we wanted to use a flag in this case, I think the flag should have been in the ViewModel since the flag is basically a logic which can be unit tested and it seems the view model has access to all the information needed to work with the flag. But it doesn't matter now :).

Reply:
Resolved in 6b0f02237bacffc9fd1e85a27af07c0c71721977

Reply:
Nitpick: We may want to consider putting the add/remove `OnPageChangeListener` methods right before and after `pager.currentItem = currentItem` to make it more explicit why we need to add/remove the listener.

Feel free to resolve this conversation without making any changes.

Reply:
I agree it is a good change, implemented in c10d46c24cfbf314574c1bf4106698f9c4949878